### PR TITLE
Homework 5

### PR DESCRIPTION
**Pull request by CjVizer**
OMG!!! xD

### CODE REVIEW

File: Homework_5/Task_4/task_4.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            # All values in the list "time_range" are seconds.
+
+# In this part of the code, we format the lines and write them to a new file
+with open('statistic_modified.txt', 'w', encoding='utf-8') as statistic_modified:
+    for item in db:
+        statistic_modified.write(f"IP: {item:>15}  "
+                                 f"Total visits: {db[item]['total_visits']:>2}  "
+                                 f"Most visited day: {most_visited_day(db[item]['visit_days'])}\n")
+    statistic_modified.write(f'Most visited time: '
+                             f'{str(timedelta(seconds=time_range[0]))} - {str(timedelta(seconds=time_range[-1]))}')
```
Comment (katevoskresenska) [inline Homework_5/Task_4/task_4.py:64]: All is good. The comments are really helpful and make the code more readable and understandable, thanks for that. One comment from me - I would add functions to your pieces of code, like: 
def remove_words_from_3_to_5_chars(in_file, out_file):
and then use your logic blocks.

File: Homework_5/Task_1/task_1.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from string import punctuation
+
+"""
+Из текстового файла удалить все слова, содержащие от трех до пяти символов,
+но при этом из каждой строки должно быть удалено только четное количество таких слов.
+"""
+
+with open('poem.txt', 'r', encoding='utf-8') as file_original:  # Opening the original file
```
Comment (mmykhailenko) [inline Homework_5/Task_1/task_1.py:8]: if you have mode='r' then you should check if the file exists or not, before you try to open it.

File: Homework_5/Task_4/task_4.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        if list_of_days.count(value) > list_of_days.count(result):
+            result = value
+    return result
+
+
+# In this part of the code, we create a structured database for our needs.
+with open('statistic.txt', 'r', encoding='utf-8') as statistic:
+    for line in statistic:  # Reading data from a file line by line then we cut off the line break,
+        ip, time, weekday = line.strip('\n').split()  # split the line with spaces, and write the values to variables.
+        if not db.get(ip):  # If key[ip] is not in the database
```
Comment (mmykhailenko) [inline Homework_5/Task_4/task_4.py:28]: db.get(ip) will return None, better to be more specific and check `If db.get(ip) is not None:` or `if not db.get(ip, '')`

File: Homework_5/Task_4/task_4.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            db[ip] = {'total_visits': 1, 'visit_days': [weekday]}  # initialize key - value
+        else:  # else increase "total_visit" by 1 and add the day of the week to the visit_day list
+            db[ip]['total_visits'] += 1
+            db[ip]['visit_days'].append(weekday)
+            # Convert time to seconds and add the result to the "timeline" list
+        time_to_seconds = list(map(int, time.split(':')))  # Variable for storing values "hours", "minutes", "seconds"
+        timeline.append(time_to_seconds[0] * 3600 + time_to_seconds[1] * 60 + time_to_seconds[2])
```
Comment (mmykhailenko) [inline Homework_5/Task_4/task_4.py:35]: better move this code in function

**Review by mmykhailenko — APPROVED** [review]

**Review by katevoskresenska — APPROVED** [review]

File: Homework_5/Task_4/task_4.py (hunk 1 above)
Comment (CjVizer) [reply Homework_5/Task_4/task_4.py]: Done )

File: Homework_5/Task_1/task_1.py (hunk 2 above)
Comment (CjVizer) [reply Homework_5/Task_1/task_1.py]: Fixed! )

File: Homework_5/Task_4/task_4.py (hunk 3 above)
Comment (CjVizer) [reply Homework_5/Task_4/task_4.py]: Fixed)

File: Homework_5/Task_4/task_4.py (hunk 4 above)
Comment (CjVizer) [reply Homework_5/Task_4/task_4.py]: Done

**Review by mmykhailenko — APPROVED** [review]
much better

**Pull request merged**